cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/DefaultAdmin.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.app.runtime;
+
+import co.cask.cdap.api.Admin;
+import co.cask.cdap.api.dataset.DatasetManagementException;
+import co.cask.cdap.api.dataset.DatasetProperties;
+import co.cask.cdap.api.dataset.DatasetSpecification;
+import co.cask.cdap.data.dataset.SystemDatasetInstantiator;
+import co.cask.cdap.data2.dataset2.DatasetFramework;
+import co.cask.cdap.proto.Id;
+import co.cask.cdap.proto.id.NamespaceId;
+
+import java.io.IOException;
+import javax.annotation.Nullable;
+
+/**
+ * Implementation of Admin that delegates dataset operations to a dataset framework.
+ */
+public class DefaultAdmin implements Admin {
+
+  private final DatasetFramework dsFramework;
+  private final NamespaceId namespace;
+  private final SystemDatasetInstantiator instantiator;
+
+  public DefaultAdmin(DatasetFramework dsFramework, NamespaceId namespace,
+                      SystemDatasetInstantiator instantiator) {
+    this.dsFramework = dsFramework;
+    this.namespace = namespace;
+    this.instantiator = instantiator;
+  }
+
+  private Id.DatasetInstance createInstanceId(String name) {
+    return Id.DatasetInstance.from(namespace.getNamespace(), name);
+  }
+
+  @Override
+  public boolean datasetExists(String name) throws DatasetManagementException {
+    return dsFramework.getDatasetSpec(createInstanceId(name)) != null;
+  }
+
+  @Nullable
+  @Override
+  public String getDatasetType(String name) throws DatasetManagementException {
+    DatasetSpecification spec = dsFramework.getDatasetSpec(createInstanceId(name));
+    return spec == null ? null : spec.getType();
```

Comment:
I implemented this according to the design that was reviewed, But you are right, this is more consistent with getDataset(), which also throws an exception when the dataset is not found. Will change it.